cmd/registry/scoring/parser_test.go

```diff
@@ -1389,3 +1408,138 @@ func TestValidateBooleanThresholds(t *testing.T) {
+					"$resource.spec/artifacts/score-lang-reuse",
+					"$resource.spec/artifacts/score-security-audit",
+					"$resource.spec/artifacts/score-accuracy",
+				},
+			},
+			wantNumErr: 1,
+		},
+		{
+			desc:   "missing score_patterns",
+			parent: "projects/demo/locations/global",
+			scoreCardDefinition: &rpc.ScoreCardDefinition{
+				Id:   "test-scorecard-definition",
+				Kind: "ScoreCardDefinition",
+				TargetResource: &rpc.ResourcePattern{
+					Pattern: "apis/-/versions/-/specs/-",
+				},
+				// error
+			},
+			wantNumErr: 1,
+		},
+		{
+			desc:   "invalid target_resource and missing score_patterns",
+			parent: "projects/demo/locations/global",
+			scoreCardDefinition: &rpc.ScoreCardDefinition{
+				Id:   "test-scorecard-definition",
+				Kind: "ScoreCardDefinition",
+				TargetResource: &rpc.ResourcePattern{
+					Pattern: "apis/-/versions/specs/-", //error
+				},
+				//error
+			},
+			wantNumErr: 2,
+		},
+		{
+			desc:   "invalid score_pattern $resource",
+			parent: "projects/demo/locations/global",
+			scoreCardDefinition: &rpc.ScoreCardDefinition{
+				Id:   "test-scorecard-definition",
+				Kind: "ScoreCardDefinition",
+				TargetResource: &rpc.ResourcePattern{
+					Pattern: "apis/-/versions/-/specs/-",
+				},
+				ScorePatterns: []string{
+					"$resource.specs/artifacts/score-lint-error", //error
+					"$resource.spec/artifacts/score-lang-reuse",
+					"$resource.spec/artifacts/score-security-audit",
+					"$resource.spec/artifacts/score-accuracy",
+				},
+			},
+			wantNumErr: 1,
+		},
+		{
+			desc:   "invalid score_pattern no artifactID",
+			parent: "projects/demo/locations/global",
+			scoreCardDefinition: &rpc.ScoreCardDefinition{
+				Id:   "test-scorecard-definition",
+				Kind: "ScoreCardDefinition",
+				TargetResource: &rpc.ResourcePattern{
+					Pattern: "apis/-/versions/-/specs/-",
+				},
+				ScorePatterns: []string{
+					"$resource.spec/artifacts/-", //error
+				},
+			},
+			wantNumErr: 1,
+		},
+		{
+			desc:   "multiple invalid score_pattern",
+			parent: "projects/demo/locations/global",
+			scoreCardDefinition: &rpc.ScoreCardDefinition{
+				Id:   "test-scorecard-definition",
+				Kind: "ScoreCardDefinition",
+				TargetResource: &rpc.ResourcePattern{
+					Pattern: "apis/-/versions/-/specs/-",
+				},
+				ScorePatterns: []string{
+					"$resource.specs/artifacts/score-lint-error",     //error
+					"$resource.specs/artifacts/score-lang-reuse",     //error
+					"$resource.specs/artifacts/score-security-audit", //error
+					"$resource.specs/artifacts/score-accuracy",       //error
+				},
+			},
+			wantNumErr: 4,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			gotErrs := ValidateScoreCardDefinition(ctx, test.parent, test.scoreCardDefinition)
+			if len(gotErrs) != test.wantNumErr {
+				t.Errorf("ValidateScoreCardDefinition() returned unexpected no. of errors: want %d, got %s", test.wantNumErr, gotErrs)
```

Comment:
This error message could include the function arguments to be more informative.

Reply:
Done.